mypy/test/testcheck.py

```diff
@@ -23,81 +24,16 @@
 
 
 # List of files that contain test case descriptions.
-typecheck_files = [
-    'check-basic.test',
-    'check-union-or-syntax.test',
-    'check-callable.test',
-    'check-classes.test',
-    'check-statements.test',
-    'check-generics.test',
-    'check-dynamic-typing.test',
-    'check-inference.test',
-    'check-inference-context.test',
-    'check-kwargs.test',
-    'check-overloading.test',
-    'check-type-checks.test',
-    'check-abstract.test',
-    'check-multiple-inheritance.test',
-    'check-super.test',
-    'check-modules.test',
-    'check-typevar-values.test',
-    'check-unsupported.test',
-    'check-unreachable-code.test',
-    'check-unions.test',
-    'check-isinstance.test',
-    'check-lists.test',
-    'check-namedtuple.test',
-    'check-narrowing.test',
-    'check-typeddict.test',
-    'check-type-aliases.test',
-    'check-ignore.test',
-    'check-type-promotion.test',
-    'check-semanal-error.test',
-    'check-flags.test',
-    'check-incremental.test',
-    'check-serialize.test',
-    'check-bound.test',
-    'check-optional.test',
-    'check-fastparse.test',
-    'check-warnings.test',
-    'check-async-await.test',
-    'check-newtype.test',
-    'check-class-namedtuple.test',
-    'check-selftype.test',
-    'check-python2.test',
-    'check-columns.test',
-    'check-functions.test',
-    'check-tuples.test',
-    'check-expressions.test',
-    'check-generic-subtyping.test',
-    'check-varargs.test',
-    'check-newsyntax.test',
-    'check-protocols.test',
-    'check-underscores.test',
-    'check-classvar.test',
-    'check-enum.test',
-    'check-incomplete-fixture.test',
-    'check-custom-plugin.test',
-    'check-default-plugin.test',
-    'check-attr.test',
-    'check-ctypes.test',
-    'check-dataclasses.test',
-    'check-final.test',
-    'check-redefine.test',
-    'check-literal.test',
-    'check-newsemanal.test',
-    'check-inline-config.test',
-    'check-reports.test',
-    'check-errorcodes.test',
-    'check-annotated.test',
-    'check-parameter-specification.test',
-    'check-generic-alias.test',
-    'check-typeguard.test',
-    'check-functools.test',
-    'check-singledispatch.test',
-    'check-slots.test',
-    'check-formatting.test'
-]
+test_path = Path(os.getcwd())
+test_path = Path(test_path).parents[1]
+test_path = test_path / Path('test-data')
+test_path = test_path / Path('unit')
```

Comment:
- First thing here accessing current working directory.parents and then the 2nd item in that seems incredibly unreliable as if someone doesn't have this in a specific level on their machine this will fail. 
- Secondly `Path.__truediv__` supports using strings directly instead of manually creating an instance of `Path`. 
- Thirdly assigning to test_path here multiple times doesn't make much sense you should just put this all on one line like:
```py
test_path = Path.cwd().parent.parent / 'test-data' / 'unit'
```

Reply:
Ok I will make the necessary changes. 
Also, will a user have different paths to the test files? in that case, can I simply set the test path to the root directory, so that all test files starting with 'check-' are detected, regardless of where they are stored?

Reply:
No if someone has moved the tests from the mypy root that's their fault and shouldn't be something you can account for

Reply:
Ok, noted.
I have committed the necessary changes, please do review them @Gobot1234 
Thanks for the cooperation!

Reply:
It's looking much better, good job! Last thing would be making sure that the version conditional tests for 3.8 and 3.9 are correctly appended (currently they are both going to be there even if the features are unsupported as far as I can tell)